components/modal/src/ModalHeader.js

```diff
@@ -0,0 +1,54 @@
+/** @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { CloseIcon } from '@westpac/icon';
+import { Title } from './styled';
+
+// ==============================
+// Component
+// ==============================
+export const ModalHeader = ({ id, onClose, children, ...props }) => {
+	const { modal, colors } = useTheme();
+
+	return (
+		<div
+			css={{
+				display: 'flex',
+				justifyContent: 'space-between',
+				alignItems: 'center',
+				overflow: 'hidden',
+				borderBottom: `${modal.header.borderWidth} solid ${modal.header.borderColor}`,
+				padding: modal.header.padding,
+			}}
```

Comment:
Button needs to sit at a consistent top/right position. Check out the small modal example. The button is moving down.

You could `alignItems: flex-start` it, or `position: absolute` it. Need to make sure that no matter what heading text is in there the button doesnt move, and the text cant show underneath the button.

Reply:
_see comment below_

Put `alignItems: flex-start` on the flex parent (this ModalHeader wrapping the flex children) so both the title and the button are top aligned.